modules/ui/src/main/scala/lucuma/ui/utils/UAParser.scala

```diff
@@ -14,8 +14,25 @@ trait Browser extends js.Object {
   val version: String = js.native
 }
 
-@js.native
-@JSImport("ua-parser-js", JSImport.Namespace)
-class UAParser(val ua: String) extends js.Object {
-  def getBrowser(): Browser = js.native
+sealed trait UAParser extends js.Object {
+  def getBrowser(): Browser
+}
+
+object UAParser {
+  def apply(ua: String): UAParser =
+    if (scala.scalajs.runtime.linkingInfo.productionMode) new UAParserProd(ua)
+    else new UAParserDev(ua)
+
+  @js.native
+  @JSImport("ua-parser-js", JSImport.Namespace)
+  class UAParserDev(val ua: String) extends js.Object with UAParser {
```

Comment:
Small thing: `UAParser` already extends `js.Object`.

Does it work if we declare `= js.native` in the `UAParser` trait, so that we don't have to do it twice in the classes?

Reply:
Good question, I don't know if that would work though somehow I doubt it. Worth testing